plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,84 @@
+#include "nlp_eng.hpp"
+#include "nlp_engine.h"
+
+#ifdef _WIN32
+#include<direct.h>
+#else
+#include <unistd.h>
+#endif
+#include <filesystem>
+using namespace std;
+
+#define NLP_DEBUG
+#ifdef NLP_DEBUG
+#include <iostream>
+#define NLP_DEBUG_FILE "/tmp/nlp_debug.txt"
+#endif
+static NLP_ENGINE *nlpEngine = NULL;
+
+NLPEng::NLPEng(const char *workingFolderPath) {
+
+#ifdef NLP_DEBUG
+    ofstream handle;
+    handle.open(NLP_DEBUG_FILE, std::ofstream::out | std::ofstream::app);
+#endif
+
+    if (workingFolderPath[0]) {
+        _stprintf(workingFolder,"%s/%s",workingFolderPath,_T("nlp-engine"));
+
+    } else {
+        char working[FILENAME_MAX+1];
+        readlink("/proc/self/exe", working, FILENAME_MAX);
+
+#ifdef NLP_DEBUG        
+        handle << _T("[readlink: ") << working << _T("]") << endl;
+#endif
+        string w = working;
+        size_t pos = w.find_last_of(_T("/"));
+        size_t pos2 = w.find_last_of(_T("/"),pos-1);
+        string parent = w.substr(0,pos2);
+        _stprintf(workingFolder,"%s/%s",parent.c_str(),_T("plugins/nlp/nlp-engine"));
```

Comment:
`workingFolder = parent+ "/plugins/nlp/nlp-engine";` would be simpler (would need workingFolder to be a std::string)?